include/flexflow/simulator.h

```diff
@@ -50,11 +50,22 @@ class FFModel;
 
 struct CostMetrics {
   CostMetrics()
-      : forward_time(0.0f), backward_time(0.0f), sync_time(0.0f),
-        memory_requirement(0) {}
+      : forward_time{}, backward_time{}, sync_time{}, inputs_memory{},
+        outputs_memory{}, weights_memory{} {}
+
+  inline size_t total_memory() const {
```

Comment:
Why `inline`?

Reply:
Also it might be nicer to just make this a function as opposed to a method

Reply:
Not really. We need the information of the object. So it's better to make it part of the object. It's not memory intensive so we don't need to save the extra memory caused by the method pointer.

Reply:
Btw, inline won't make difference, because the compiler should optimize it. But it's better to guide the compiler to do so.